Ascend-Server.api/Controllers/v1/GuidedJournalEntryController.cs

```diff
@@ -0,0 +1,166 @@
+using Data.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using AutoMapper;
+using Dto;
+using IServices;
+using Ascend_Server.api.ActionFilters;
+using Microsoft.AspNetCore.Authorization;
+using Services;
+
+namespace Controllers;
+
+[Authorize]
+[ApiController]
+[Route("api/v{version:apiVersion}/[controller]")]
+[ServiceFilter(typeof(ModelStateActionFilter))]
+[ApiVersion("1.0")]
+public class GuidedJournalEntryController : ControllerBase
+{
+    private readonly IGuidedJournalEntryService _guidedJournalEntryService;
+
+    private readonly IMapper _mapper;
+
+    public GuidedJournalEntryController(IGuidedJournalEntryService guidedJournalEntryService,
+                                        IMapper mapper)
+    {
+        _guidedJournalEntryService = guidedJournalEntryService;
+
+        _mapper = mapper;
+    }
+    /// <summary>
+    /// Retrieves a single Guided Journal Entry that matches the passed id.
+    /// </summary>
+    /// <returns>
+    /// Returns a single instance of type <see cref="Dto.GuidedJournalEntry"/> object matching the id passed.
+    /// </returns>
+    /// <response code="200">Returns the Guided Journal Entry.</response>
+    /// <response code="400">The request was invalid.</response>
+    /// <response code="404">No guided journal entry found for id.</response>
+    /// 
+    [HttpGet("id/{id}")]
+    public ActionResult<Dto.GuidedJournalEntry> GetById(Guid id)
+    {
+        try
+        {
+            var guidedJournalEntry = _guidedJournalEntryService.GetById(id);
+
+            var dto = _mapper.Map<Data.GuidedJournalEntry, Dto.GuidedJournalEntry>(guidedJournalEntry);
+
+            return new OkObjectResult(dto);
+        }
+        catch (NotFoundException e)
```

Comment:
You'll also want to catch general exceptions and not send back a detailed error message so that you don't leak sensitive information. Follow this practice in all controller actions.